### ISSUE TITLE
0.47.7 MekHQ-MegaMek Princess halting during move, NPE 

### ISSUE THREAD

**Issue by gsparks3** [MegaMek/megamek#2115]
### Environment
Version: 0.47.7
MekHQ integrated MegaMek
Windows 10
Java version 1.8.0_261

### Description
Princess locked up while attempting to move (possibly related to ejected Mechwarrior in water being the active unit?). Kicked and replaced bot, immediately locked up again, at which point I started taking logs and screenshots, etc. Uncertain if previous freeze was identical, but it shows two of the same NPE in MekHQlog.txt, while megameklog.txt does not have anything immediately obvious to me. Included both anyway. 
![image](https://user-images.githubusercontent.com/59574316/89468925-76db7d80-d746-11ea-8b8e-d4cda936e511.png)

Edit: after testing, issue reproduces  after save/quit/load and replacing all bots. Kicking the Lyran Commonwealth bot, skipping a single turn, then replacing the bot resolves the issue (at least temporarily), so it does seem linked to that specific unit. 

### Files
[megameklog.txt](https://github.com/MegaMek/megamek/files/5031483/megameklog.txt)
[mekhqlog.txt](https://github.com/MegaMek/megamek/files/5031495/mekhqlog.txt)
[princessNPE.sav.gz](https://github.com/MegaMek/megamek/files/5031493/princessNPE.sav.gz)

Player name: Coppercoats

**Comment by HammerGS** [comment]
NPE from log
```
java.lang.NullPointerException
	at megamek.common.pathfinder.PathDecorator.decoratePath(PathDecorator.java:66)
	at megamek.client.bot.princess.Princess.getMovePathsAndSetNecessaryTargets(Princess.java:1574)
	at megamek.client.bot.princess.Princess.continueMovementFor(Princess.java:1341)
	at megamek.client.bot.BotClient.calculateMyTurn(BotClient.java:517)
	at megamek.client.bot.BotClient.access$000(BotClient.java:85)
	at megamek.client.bot.BotClient$CalculateBotTurn.run(BotClient.java:107)
	at java.lang.Thread.run(Unknown Source)
```

**Comment by gsparks3** [comment]
Update: Loading from save, force-skipping the affected unit's turn and playing through the round of combat is consistently resulting in _another_ NPE when attempting to move to the Initiative phase of the next round. See attached save and logs (reloaded again, repeated, and saved immediately before pressing Done, which triggers the issue). This one appears to be a NPE from a different part of Megamek - should it get its own issue? General environment details are otherwise the same. 

[mekhqlog.txt](https://github.com/MegaMek/megamek/files/5031608/mekhqlog.txt)
[megameklog.txt](https://github.com/MegaMek/megamek/files/5031611/megameklog.txt)
[princessNPE2.sav.gz](https://github.com/MegaMek/megamek/files/5031612/princessNPE2.sav.gz)

**Comment by HammerGS** [comment]
> Update: Loading from save, force-skipping the affected unit's turn and playing through the round of combat is consistently resulting in _another_ NPE when attempting to move to the Initiative phase of the next round. See attached save and logs (reloaded again, repeated, and saved immediately before pressing Done, which triggers the issue). This one appears to be a NPE from a different part of Megamek - should it get its own issue? General environment details are otherwise the same.

NPE from this Section
```
Exception in thread "Packet Pump" java.lang.NullPointerException
	at megamek.common.Player.getCommandBonus(Player.java:528)
	at megamek.common.Team.getTotalInitBonus(Team.java:319)
	at megamek.common.TurnOrdered.rollInitAndResolveTies(TurnOrdered.java:376)
	at megamek.common.TurnOrdered.rollInitiative(TurnOrdered.java:290)
	at megamek.server.Server.rollInitiative(Server.java:3425)
	at megamek.server.Server.prepareForPhase(Server.java:2274)
	at megamek.server.Server.changePhase(Server.java:2227)
	at megamek.server.Server.endCurrentPhase(Server.java:3020)
	at megamek.server.Server.checkReady(Server.java:2019)
	at megamek.server.Server.handle(Server.java:31572)
	at megamek.server.Server$PacketPump.run(Server.java:259)
```

**Comment by NickAragua** [comment]
The first issue is fixed in 47.8.
